applications/fmc/src/fmc_route_req.erl

```diff
@@ -0,0 +1,88 @@
+%%%-------------------------------------------------------------------
+%%% @copyright (C) 2013-2015, 2600Hz, INC
+%%% @doc
+%%%
+%%% @end
+%%% @contributors
+%%%   SIPLABS, LLC (Vladimir Potapev)
+%%%-------------------------------------------------------------------
+-module(fmc_route_req).
+
+-export([handle_req/2]).
+
+-include("fmc.hrl").
+
+handle_req(JObj, Props) ->
+    lager:info("FMC tried to handle route_req"),
+    lager:debug("FMC JObj is ~p", [JObj]),
+    lager:debug("FMC Props is ~p", [Props]),
+    'true' = wapi_route:req_v(JObj),
+    CallId = wh_json:get_value(<<"Call-ID">>, JObj),
+    put('callid', CallId),
+    Call = whapps_call:from_route_req(JObj),
+    
+    ANumber = wh_json:get_value(<<"Caller-ID-Number">>, JObj),
+    FMCHeaderValue = wh_json:get_value([<<"Custom-SIP-Headers">>, ?XFMC_HEADER], JObj),
+    FMCHeaderValueRegexp = ?XFMC_REGEXP,
+    FMCHeaderValue1 = binary_to_list(FMCHeaderValue),
+    FMCHeaderValueRegexp1 = binary_to_list(FMCHeaderValueRegexp),
+    ExtractedFMCValue = case re:run(FMCHeaderValue1, FMCHeaderValueRegexp1) of
+                            'nomatch' -> 'undefined';
+                            {'match', Groups} ->
+                                {Pos, Len} = lists:nth(2, Groups),
+                                ExtractedValue = lists:sublist(FMCHeaderValue1, Pos + 1, Len),
+                                list_to_binary(ExtractedValue)
```

Comment:
Look into the options to `re:run/3` as you can ask it to `{return, binary}` and `match_only` (or in another wording).